### PR TITLE
Transformations: Various enhancements & fixes

### LINKED ISSUE #1844: Add option to copy UID for transformation
When creating a transformation in Main UI a UID like `config:map:6567df89` is created. It would be nice to have a `copy` button in the details view for easier handling.

### PR DESCRIPTION
**Pull request by florian-h05**
Closes #1844.

* Allows to copy the UID.
* Sorts the transformation types in the "by types" list alphabetically and makes the segment labels uppercase to match the transformation service names.
* Adjusts to recent core change https://github.com/openhab/openhab-core/pull/3487: removes the script language selection and updates the editor mode and code snippet injection.
* Extends language support and refactors the translate mode logic in `script-editor.vue`.
* Fixes the clear label button not displayed in create mode and displayed for non-editable transformations.
* Shows a tip on how to use the transformation for Item states.
* Fixes failing and not required API request after transformation creation.
* Removes possibility to select and attempt to delete not-editable transformations, which is not possible and failed.

### CODE REVIEW

**Comment by florian-h05** [comment]
/cc @J-N-K @rkoshak

@ghys Would it be possible to get this reviewed and merged until 4pm to have this in the second milestone?

File: bundles/org.openhab.ui/web/src/assets/definitions/transformations.js [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
+    XSLT: '<?xml version=\'1.0\' encoding=\'UTF-8\'?>\n' +
+      '<xsl:stylesheet version=\'2.0\' xmlns:xsl=\'http://www.w3.org/1999/XSL/Transform\'>\n' +
+      '   <xsl:output method=\'xml\' indent=\'no\'/>\n' +
+      '   <xsl:template match=\'/\'>\n' +
+      '      <reRoot><reNode><xsl:value-of select=\'/root/node/@val\' /> world</reNode></reRoot>\n' +
+      '   </xsl:template>\n' +
+      '</xsl:stylesheet>'
   },
   EDITOR_MODES: {
+    DSL: 'application/vnd.openhab.dsl.rule',
```
Comment (J-N-K) [inline bundles/org.openhab.ui/web/src/assets/definitions/transformations.js:18]: Do we need `JS` here? Otherwise LGTM.

Comment (florian-h05) [reply bundles/org.openhab.ui/web/src/assets/definitions/transformations.js:18]: No, we don't.

When a transformation is loaded, the UI checks if there is an editor mode defined in this file. If it is there this is passed to `script-editor.vue`, if it is not there, the transformation type is used.
The UI's code editor (codemirror) supports various language names out of the box, e.g. xml, and `script-editor.vue` translates language shortcuts such as js, py, rb to support them out of the box.

The `EDITOR_MODES` definitions are only required for transformation services where the name does not securely indicate the MIME type or language name-

**Review by J-N-K — APPROVED** [review]

**Review by ghys — APPROVED** [review]
I'll be on a plane by 4pm but overall LGTM at first sight. Feel free to self merge when you're done with the last fixes and the checks pass!

**Pull request merged**